src/expire.c

```diff
@@ -218,6 +218,7 @@ static long long activeExpireCycleJob(enum activeExpiryType jobType, int cycleTy
     int j, iteration = 0;
     int dbs_per_call = CRON_DBS_PER_CALL;
     int dbs_performed = 0;
+    int time_check_rate; /* Check time limit every 1/Xth of the loop. */
```

Comment:
Looking at the code below seeing 0x0, I thought it will be a division by zero, but it's actually this comment being slightly off.

We use it as a mask. We check the time every X + 1 iteration, to be exact.
Should we call it mask instead?

```suggestion
    int time_check_mask; /* Check time limit when (i & mask) == 0, i.e. every (X+1)th of the loop. */
```